src/test/ui/consts/const_short_circuit.stderr

```diff
@@ -0,0 +1,26 @@
+error: new features like let bindings are not permitted in constant which also use short circuiting operators
+  --> $DIR/const_short_circuit.rs:6:9
+   |
+LL |     let mut x = true && false;
```

Comment:
also, I presume we intend this to be temporary? this restriction is very odd and surprising from an end-user point-of-view, I wonder if we can direct people to some kind of roadmap / URL that explains what the heck is going on?